packages/flutter/lib/src/widgets/editable_text.dart

```diff
@@ -712,7 +727,15 @@ class EditableText extends StatefulWidget {
        assert(rendererIgnoresPointer != null),
        assert(scrollPadding != null),
        assert(dragStartBehavior != null),
+       assert(
+          (contentInsertionMimeTypes.isNotEmpty && onContentInserted != null) ||
+          contentInsertionMimeTypes.isEmpty,
+          'onContentInserted cannot be null if contentInsertionMimeTypes is provided',
+       ),
        enableInteractiveSelection = enableInteractiveSelection ?? (!readOnly || !obscureText),
+       contentInsertionMimeTypes = onContentInserted != null && contentInsertionMimeTypes == const <String>[]
```

Comment:
Maybe just set the default to `kDefaultContentInsertionMimeTypes` instead of an empty List? Changing the allowed mime types based on whether the list is empty can be surprising.

Reply:
This list of mime types determines whether the image content can be inserted or not inside the engine. 

The reason I keep it as an empty list at first is because if `onContentInserted` is not provided, I want to fully disable the ability to insert content. This is especially important because the OS will show a toast message indicating that the text field does not support inserting content. 

If I kept the default as kDefaultMimeTypes then the OS would think that the content is allowed to be inserted (even though the dev didn't provide any handler for it), and the user might be confused because the toast message doesn't show, yet nothing happened on the app.

Reply:
You can do that later / at a lower level right? For example when constructing the TextInputConfiguration. People would think setting this parameter to an empty list would disable this feature and would not expect this to be set back to `kDefaultMimeTypes` when they set this to an empty list.

Reply:
Also could you add it to the documentation about how this can be disabled?

Reply:
I guess that's fair, although if `onContentInserted` is supplied I wouldn't think that people would want to disable the feature? I can see how this is a bit confusing, to be completely honest I'm not sure the best API design here.

Reply:
@justinmc do you have any feedback on the API design here? I really don't have much idea on how I can make it any better. I'll try and outline all the possible cases of a dev setting `onContentInserted` and `contentInsertionMimeTypes` (abbreviated to OCI and CIMT) in their current state here:

1. OCI null and CIMT provided as not empty list - caught by assert. Not allowed as you should provide a handler for inserted content if you plan on accepting mime types.

2. OCI not null and CIMT provided

   1. CIMT provided as empty list - CIMT reset back to default mime types. The assumption here is that a dev would want to accept content insertion if they've provided a handler, so let's use the default mime types.
   
   2. CIMT provided as not empty list - CIMT not modified and sent as-is to engine.
   
3. OCI not null and CIMT not provided - CIMT reset back to default mime types.

For a dev to disable content insertion, they must set OCI as null and provide CIMT as an empty list or null. 

Does this seem like the correct approach? I've been thinking about @LongCatIsLooong 's request and I'm not sure I can make that happen.

Reply:
I don't think there's a silver bullet solution here, but here are some ideas.  I agree with @LongCatIsLooong that changing a passed in `[]` to kDefaultCIMT is probably not the right thing to do.

  1. contentInsertionMimeTypes defaults to `[]`.
    - ❌ We can't distinguish between if the user passes nothing or `[]`.  This is probably a no-go because of the case where the user passes in OCI.  If they pass nothing for CIMT, they want it to default to kDeafultCIMT, but if they explicitly pass `[]`, there should probably be an assertion.
  2. contentInsertionMimeTypes defaults to `kDefaultContentInsertionMimeTypes`.
    - ⚠️ We can't handle the situation where the user passes kDefaultCIMT when OCI is null.  That should be an assertion.
  3. contentInsertionMimeTypes defaults to `null`.
    - ⚠️ We can't distinguish between passing `null` and not passing anything.  This is a problem when the user sets OCI and explicitly passes `null` to CIMT, which should probably be an assertion.

None of these is ideal, but maybe no. 2 isn't so bad?  Otherwise we could look at fancier solutions:

  4. Pass both of these parameters via a single ContentInsertionConfiguration parameter, where that new class takes both as parameters.
    - This increases complexity but might be able to solve the problems of the above solutions.  It could be worth it but it's not clear to me without trying it.
  5. Use the sentinel pattern.
    - A few places in Flutter/Dart, we set the parameter default to a unique, nonsensical value in order to distinguish between passed null and not passed in the initializers, where the actual default value can be set.  This is confusing so it might be overkill here.

Maybe no. 4 is worth investigating if you're not happy with no. 2, or what do you think?

Reply:
I really like how no. 4 turned out. Adding the extra layer of the configuration class allows me to achieve the result I wanted without making the API seem confusing and counter-intuitive. Thanks for the idea!